lib/libzrepl/data_conn.c

```diff
@@ -1436,6 +1458,8 @@ uzfs_zvol_rebuild_scanner(void *arg)
 					    ZVOL_OPCODE_REBUILD_ALL_SNAP_DONE,
 					    fd, NULL, 0, 0);
 					all_snap_done = B_TRUE;
+					wait_snap_timeout();
+					goto snap_reverify;
```

Comment:
We planned to break the rebuild connection if any snapshot commands comes during the wait time.

Reply:
we are breaking if there is a snapshot in snap_reverify step as snap_zv will not be null in that case but all_snap_done will be true.

Reply:
sounds good

Reply:
this will have a problem, as, the snap_zv will be destroyed if all_snap_done is true.

Reply:
yes.. at the `exit`, there is code that destroys snap_zv if all_snap_done is true. Probably, you may need to sync your code to upstream.

Reply:
that is ok, as it will destroy internal snapshot.

Reply:
ok, in case of user snapshot snapsshot, it will destroy that also. will fix it.